homeassistant/components/python_script/__init__.py

```diff
@@ -146,6 +149,36 @@ def python_script_service_handler(call: ServiceCall) -> ServiceResponse:
         async_set_service_schema(hass, DOMAIN, name, service_desc)
 
 
+IOPERATOR_TO_OPERATOR = {
+    "%=": operator.mod,
+    "&=": operator.and_,
+    "**=": operator.pow,
+    "*=": operator.mul,
+    "+=": operator.add,
+    "-=": operator.sub,
+    "//=": operator.floordiv,
+    "/=": operator.truediv,
+    "<<=": operator.lshift,
+    ">>=": operator.rshift,
+    "@=": operator.matmul,
+    "^=": operator.xor,
+    "|=": operator.or_,
+}
+
+
+def guarded_inplacevar(op: str, target: Any, operand: Any) -> Any:
+    """Implement augmented-assign (+=, -=, etc.) operators for restricted code.
+
+    See RestrictedPython's `visit_AugAssign` for details.
+    """
+    if not isinstance(target, (list, Number, str)):
+        raise ScriptError(f"The {op!r} operation is not allowed on a {type(target)}")
```

Comment:
Add some test cases for the negative checks.

Reply:
Added a test case I could think of.

Also, I noticed `datetime.date.today()` does not work in restricted scripts because `__import__` is not available (coincidentally I'd debugged that here: https://stackoverflow.com/a/77368231/51685). Is that an issue that should be triaged? 🤔

Reply:
I don't know if anyone is really using this integration. I think most people will use the `pyscript` custom component which is far more powerful.

Reply:
Well, this PR came about by someone on a Slack wondering why `+=` didn't work, so yeah, someone is using it!